torchquad/plots/plot_adaptive_grid.py

```diff
@@ -0,0 +1,29 @@
+import matplotlib.pyplot as plt
+import numpy as np
+
+
+def plot_grid(grid, dpi=100):
+    """Plots the adaptive grid and corresponding function value.
+
+    Args:
+        grid (AdaptiveGrid): AdaptiveGrid of evaluated function
+        dpi (int, optional): Plot dpi. Defaults to 100.
+    """
+
+    fig = plt.figure(dpi=dpi)
+    points = None
+    fvals = None
+    for subdomain in grid.subdomains:
```

Comment:
I think this function only supports 2D grids, right? If so, we should raise an error if a non-2D grid is given